### PR TITLE
Provision Horde Groupware,

### PR DESCRIPTION
**Pull request by Infern1**
Mail and Ingo (mailfilter for Maildrop)

### Changes proposed in this pull request:
- Make Horde available, main use is, ingo for maildrop filters

Checklist:
- [ ] docs up-to-date

### CODE REVIEW

File: provision-horde.sh [hunk 1]
```diff
@@ -0,0 +1,490 @@
+#!/bin/sh
+
+# shellcheck disable=1091
+. mail-toaster.sh || exit
+
+export JAIL_START_EXTRA=""
+# shellcheck disable=2016
+export JAIL_CONF_EXTRA=""
+export JAIL_CONF_EXTRA="
```
Comment (msimerson) [inline provision-horde.sh]: dupe.

File: provision-horde.sh [hunk 2]
```diff
@@ -0,0 +1,490 @@
...
+    if [ ! -d "$ZFS_DATA_MNT/horde/data" ]; then
+        tell_status "creating $ZFS_DATA_MNT/horde/data"
+        mkdir "$ZFS_DATA_MNT/horde/data"
+    fi
+
+    tell_status "making vpopmail dir"
+    mkdir -p "$STAGE_MNT/usr/local/vpopmail"  || exit
+
+    install_php 56 || exit
+	install_nginx || exit
```
Comment (msimerson) [inline provision-horde.sh]: looks like you've got a mix of tabs and spaces. Usually you can configure your editor to automatically create consistent whitespace within files.

File: provision-horde.sh [hunk 3]
```diff
@@ -0,0 +1,490 @@
...
+    
+	tell_status "installing Horde IMP and Ingo "
+	stage_pkg_install horde-ingo
+    stage_pkg_install horde-imp
+    stage_pkg_install php56-simplexml php56-ftp php56-gd php56-fileinfo php56-tidy
```
Comment (msimerson) [inline provision-horde.sh]: You can add your required PHP modules to the `install_php 56` line above like so:

`install_php 56 "simplexml ftp gd fileinfo tidy"` The install_php function will take care of expanding the modules to the port names with the php version.

File: provision-horde.sh [hunk 4]
```diff
@@ -0,0 +1,490 @@
...
+    location ~ ^/horde/(.+\.php) {
+        fastcgi_split_path_info ^(.+\.php)(/.+)$;
+        fastcgi_param PATH_INFO $fastcgi_path_info;
+        fastcgi_param PATH_TRANSLATED $document_root$fastcgi_path_info;
+        fastcgi_param PHP_VALUE "cgi.fix_pathinfo=1";
+        fastcgi_pass php;
+        fastcgi_index index.php;
+        fastcgi_param SCRIPT_FILENAME $document_root$fastcgi_script_name;
+        include        /usr/local/etc/nginx/fastcgi_params;
+        root /usr/local/www/;
```
Comment (msimerson) [inline provision-horde.sh]: You've already declared root in the enclosing block. Isn't this one redundant?

File: provision-horde.sh [hunk 5]
```diff
@@ -0,0 +1,490 @@
...
+    }
+    location ~ ^/horde/(.+\.(?:ico|css|js|gif|jpe?g|png))$ {
+         root /usr/local/www/;
```
Comment (msimerson) [inline provision-horde.sh]: same.

File: provision-horde.sh (hunk 1 above)
Comment (Infern1) [inline provision-horde.sh]: will fix

File: provision-horde.sh (hunk 2 above)
Comment (Infern1) [inline provision-horde.sh]: will fix

File: provision-horde.sh (hunk 4 above)
Comment (Infern1) [inline provision-horde.sh]: getting nginx and horde to work was a nightmare... this one works...

**Comment by Infern1** [comment]
> So this adds the FTP server so that Ingo can upload the maildrop files via FTP to the locally mounted vpopmail mail store? Hmm. Ok.
Yes by using FTP it's most easy to get uid and gid right

> I suppose the alternative would be enabling ftpd in the vpopmail jail. That removes the vpopmail mount from this jail and reduces the mail store access to just via FTP. Hmm.

Yes, thought about enabling the ftp server in the vpopmail jail, but I went for this way, since I thought most people won't go for Ingo and keep the provisioned jails as independent as possible

**Comment by Infern1** [comment]
Don't know if the indenting is ok now...
Used some plugins for vim to the indentation...

**Comment by msimerson** [comment]
Just a couple settings in your ~/.vimrc file should automatically get you consistently correct automatic whitespace insertion in your shell scripts::

````
filetype plugin indent on
autocmd FileType sh,bash set noexpandtab shiftwidth=4 tabstop=4
````

**Comment by msimerson** [comment]
Looks like the only [issue left](https://travis-ci.org/msimerson/Mail-Toaster-6/builds/203988963) is adding some quotes to pass the lint tests (shellcheck), and then you should pass the CI tests.

**Comment by msimerson** [comment]
Also, if you don't already, I highly recommend using [syntastic](https://github.com/vim-syntastic/syntastic) with vim. I have it installed via pathogen so it runs the shellcheck tests on my scripts every time I save.

**Pull request merged**